petergreve/week1/ass2/index.html

```diff
@@ -0,0 +1,139 @@
+	<!-- <script src="http://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.4.4/babel.min.js"></script> -->
+	<script src="http://fb.me/JSXTransformer-0.12.2.js"></script>
+</head>
+<body>
+	<div id="root"> </div>
+
+	<!-- 1. Create an inputfield that doesn't accept any numbers -->
+	<!-- 2. Find a way to submit and print the message -->
+	<!-- 3. Validation errors in inputfields should produce an error-message rather than a hello morld message -->
+	<!-- 4. Any error-message should be printed in red color -->
+	<!-- 5. Provide a function that returns all messages ever printed -->
+	<!-- 6. This must be solved by using only native Javascript and react -->
+	<!-- 7. All markup must be produced with react -->
+	<!-- 8. All messages must be printed in the order they where submitted -->
+
+
+
+<script type="text/jsx">
+
+	var InputComponent = React.createClass({
+		  render: function() {
+		    return 	<div>
+		    			<input type="text" ref="input" />
+						<button type="button" onClick={this.validate}>Submit</button>
+		    		</div>;
+		  },
+		  validate: function() {
+		  	var message = this.refs.input.getDOMNode().value;
+		  	this.props.validate(message);
+		  }
+	});
+
+
+	var OutputComponent = React.createClass({
+		  render: function() {
+		    return <div id="message-container">
+		    			{this.props.message}
+		    		</div>;
+		  }
+	});
+
+	var ErrorMessage = React.createClass({
+		render: function () {
+
+			var style = {
+	   			color: 'red',
+    		};
+			return 	<div style={style}>
+						{this.props.errorMessage}
+					</div>
+		}
+
+	});
+
+	var AllMessages = React.createClass({
+		render: function () {
+			return 	<div>
+					    <ol>
+					   		{this.props.allMessages.map(function(message) {
+					     		return <OutputComponent message={message} />;
+					     	})}
+					    </ol>
+      				</div>
+		}
+
+	});
+
+	var DisplayAll = React.createClass({
+		  render: function() {
+		    return 	<div>
+						<button type="button" onClick={this.props.displayAll}>Display All</button>
+		    		</div>;
+		  },
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {message: 'Hello!',
+		    		validMessage: '',
+		    		errorMessage: '',
+		    		allMessages: [],
+		    		showMessages: false
+		    	};
+		},
+		render: function() {
+			return 	<div>
+						<InputComponent validate={this.validate} />
+						<DisplayAll displayAll={this.displayAllMessages} />
+						<OutputComponent message={this.state.validMessage} />
+						<ErrorMessage errorMessage={this.state.errorMessage} />
+						{ this.state.showMessages ? <AllMessages allMessages={this.state.allMessages} /> : null }
+					</div>
+		},
+		validate: function(message) {
+			if (!/^[0-9]*$/.test(message)) {
+				this.setState({
+		          validMessage: message,
+		          errorMessage: '',
+		          allMessages: this.state.allMessages.concat([message]),
+		          showMessages: false
```

Comment:
This will result in `AllMessages` component still being visible when a validation error occurred, whilst it is hidden after a validation passes.

Is this behavior intended?